samples/subsys/llext/modules/src/hello_world_ext.c

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * This very simple hello world C code can be used as a test case for building
+ * probably the simplest loadable extension. It requires a single symbol be
+ * linked, section relocation support, and the ability to export and call out to
+ * a function.
+ */
+
+#include <zephyr/llext/symbol.h>
+#include <zephyr/sys/printk.h>
+
+void hello_world(void)
+{
+	printk("Hello, world!\n");
+}
+LL_EXTENSION_SYMBOL(hello_world);
```

Comment:
If `CONFIG_LLEXT_HELLO_WORLD=y`, this file will be built as part of the Zephyr application. However, there is nothing special in the `LL_EXTENSION_SYMBOL` macro that makes it not emit the export table entry.
Is there no risk to blow up CI with orphan section linker warnings due to something being emitted to the `.exported_sym` section which, AFAICT, is not handled in Zephyr linker script?

Reply:
Interesting... it did not in my tests, but I can't be sure it won't everywhere. I will fix this as we did for the `EXPORT_SYMBOL` macro: make it a NOP if LLEXT is not enabled, and disable LLEXT for the =y case.